File: threads/src/main/java/im/threads/internal/transport/threads_gate/ThreadsGateMessageParser.java
package im.threads.internal.transport.threads_gate;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import im.threads.internal.model.ChatItem;
import im.threads.internal.transport.MessageParser;
import im.threads.internal.transport.threads_gate.responses.BaseMessage;

final class ThreadsGateMessageParser {

    private ThreadsGateMessageParser() {
    }

    static String getType(@NonNull final BaseMessage message) {
        return MessageParser.getType(message.getContent());
    }

    @Nullable
    static ChatItem format(@NonNull final BaseMessage message) {
        return MessageParser.format(message.getMessageId(), message.getSentAt().getTime(), message.getNotification(), message.getContent());
    }

    static boolean checkId(@NonNull final BaseMessage message, final String currentClientId) {
        return MessageParser.checkId(message.getContent(), currentClientId);
    }
}

File: threads/src/main/java/im/threads/business/transport/threadsGate/ThreadsGateMessageParser.kt
package im.threads.business.transport.threadsGate

import com.google.gson.JsonObject
import im.threads.business.models.ChatItem
import im.threads.business.transport.MessageParser
import im.threads.business.transport.threadsGate.responses.BaseMessage

internal class ThreadsGateMessageParser(private val messageParser: MessageParser) {

    fun getType(message: BaseMessage): String {
        val content = message.content ?: JsonObject()
        return messageParser.getType(content)
    }

    fun format(message: BaseMessage): ChatItem? {
        return messageParser.format(
            message.sentAt?.time ?: 0L,
            message.notification,
            message.content
        )
    }
}
